hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/scale/ITestS3ADirectoryPerformance.java

```diff
@@ -137,6 +159,78 @@ public void testListOperations() throws Throwable {
     }
   }
 
+  @Test
+  public void testMultiPagesListingPerformanceAndCorrectness() throws Throwable {
+    describe("Check performance and correctness for multi page listing " +
+            "using different listing api");
+    Path dir = path(this.getMethodName());
+    Configuration conf = getConfigurationWithConfiguredBatchSize(10);
+    fs = (S3AFileSystem) FileSystem.get(dir.toUri(), conf);
```

Comment:
any reason to not use a local variable here?

Reply:
Initially there were two tests so I thought of reusing the class level variable and closing the fs in teardown. 
Since there is just one test, I have moved to a local variable and closing using try-with-resources.